modules/apps/journal/journal-service/src/test/java/com/liferay/journal/internal/util/JournalConverterImplTest.java

```diff
@@ -88,73 +96,198 @@ private void _testUpdateContentDynamicElement(
 		Assert.assertEquals(expectedValue, rootElement.getStringValue());
 	}
 
-	private void _testUpdateContentDynamicElementWithOptions() {
+	private void _testUpdateContentDynamicElementWithCheckBox() {
 		JournalConverterImpl journalConverterImpl = new JournalConverterImpl();
 
 		ReflectionTestUtil.setFieldValue(
 			journalConverterImpl, "_jsonFactory", new JSONFactoryImpl());
 
-		DDMFormField ddmFormField = _createDDMFormField(
-			"string", true, "field2", "select");
+		SAXReaderImpl saxReaderImpl = new SAXReaderImpl();
 
-		ddmFormField.setMultiple(true);
+		Document document = saxReaderImpl.createDocument();
 
-		DDMFormFieldOptions ddmFormFieldOptions = new DDMFormFieldOptions();
+		Element rootElement = document.addElement("root");
 
-		String value = RandomTestUtil.randomString();
+		boolean[] multipleFlags = {true, false};
+		List<List<String>> allOptionReferences = new ArrayList<>();
+
+		for (boolean multiple : multipleFlags) {
+			DDMFormField ddmFormField = _createDDMFormField(
+				"string", true, "checkbox_" + multiple, "checkbox_multiple");
+
+			DDMFormFieldOptions ddmFormFieldOptions = new DDMFormFieldOptions();
+
+			List<String> optionValues = new ArrayList<>();
+			List<String> optionReferences = new ArrayList<>();
+
+			for (int j = 0; j < (multiple ? 2 : 1); j++) {
+				String value = "Option" + RandomTestUtil.randomString();
+				String optionReference = RandomTestUtil.randomString();
+
+				optionValues.add(value);
+				optionReferences.add(optionReference);
+
+				ddmFormFieldOptions.addOption(value);
+				ddmFormFieldOptions.addOptionLabel(
+					value, LocaleUtil.US, RandomTestUtil.randomString());
+				ddmFormFieldOptions.addOptionReference(value, optionReference);
+			}
+
+			ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
+			allOptionReferences.add(optionReferences);
+
+			Serializable fieldValue;
+
+			if (multiple) {
+				fieldValue = JSONUtil.putAll(
+					optionValues.toArray()
+				).toString();
+			}
+			else {
+				fieldValue = optionValues.get(0);
+			}
+
+			ReflectionTestUtil.invoke(
+				journalConverterImpl, "_updateContentDynamicElement",
+				new Class<?>[] {
+					int.class, DDMFormField.class, Element.class, Field.class
+				},
+				0, ddmFormField, rootElement,
+				new Field(
+					RandomTestUtil.randomLong(), ddmFormField.getName(),
+					HashMapBuilder.<Locale, List<Serializable>>put(
+						LocaleUtil.US, ListUtil.fromArray(fieldValue)
+					).build(),
+					LocaleUtil.US));
+		}
+
+		List<Element> dynamicContentElements = rootElement.elements(
+			"dynamic-content");
 
-		ddmFormFieldOptions.addOption(value);
-		ddmFormFieldOptions.addOptionLabel(
-			value, LocaleUtil.US, RandomTestUtil.randomString());
+		Assert.assertEquals(
+			dynamicContentElements.toString(), 2,
+			dynamicContentElements.size());
 
-		String optionReference = RandomTestUtil.randomString();
+		for (int i = 0; i < dynamicContentElements.size(); i++) {
```

Comment:
Same here, a new method called "assertDynamicContentElements" and call it twice (or pass both elements)